### PR TITLE
DatastoreClusterAddDatastore DSC Resource

### PR DESCRIPTION
**Pull request by SimeonGerginov**
### Added
- Added **DatastoreClusterAddDatastore** DSC Resource.
- Added Unit Tests for **DatastoreClusterAddDatastore** DSC Resource.
- Added Integration Tests for **DatastoreClusterAddDatastore** DSC Resource.
- Added Documentation and example Configurations for **DatastoreClusterAddDatastore** DSC Resource.

### Changed
- Extended **InventoryUtil** class with **GetDatastoreCluster()**.

### CODE REVIEW

**Comment by SimeonGerginov** [comment]
Example usage of DatastoreClusterDatastore DSC Resource:

```
DatastoreClusterDatastore DatastoreClusterDatastore {
    Server = $Server
    Credential = $Credential
    DatacenterName = 'Datacenter'
    DatacenterLocation = ''
    DatastoreClusterName = 'DscDatastoreCluster'
    DatastoreClusterLocation = ''
    DatastoreNames = @('DscDatastoreOne', 'DscDatastoreTwo')
}
```

File: Source/VMware.vSphereDSC/DSCResources/DatastoreClusterDatastore.ps1 [hunk 1]
```diff
@@ -0,0 +1,280 @@
...
+                    $this.DatastoreClusterLocation,
+                    $datacenter,
+                    $datacenterDatastoreFolderName
+                )
+            )
+
+            $datastores = $this.GetDatastores($datacenter)
+            $datastoresToAddToDatastoreCluster = $this.GetDatastoresToAddToDatastoreCluster($datastoreCluster, $datastores)
+
+            $this.AddDatastoresToDatastoreCluster($datastoreCluster, $datastoresToAddToDatastoreCluster)
```
Comment (dmilov) [inline Source/VMware.vSphereDSC/DSCResources/DatastoreClusterDatastore.ps1:101]: Why only Add is applied in this resource?
I my intent is to configure the datastore in a datastore cluster with this resource and I specify the list of desired datastores I'd expect to have the exact those after Set configuration. It seems this resource is just for the purpose of adding datastore.

File: Documentation/DSCResources/DatastoreClusterDatastore/DatastoreClusterDatastore.md [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+| **DatastoreClusterName** | Key | string | The name of the Datastore Cluster located in the Datacenter specified in **DatacenterName** key property. ||
+| **DatastoreClusterLocation** | Key | string | The location of the Datastore Cluster with name specified in **DatastoreClusterName** key property in the Datacenter specified in **DatacenterName** key property. Location consists of 0 or more Folders. Empty location means that the Datastore Cluster is located in the Datastore Folder of the Datacenter. The Root Folders of the Datacenter are not part of the location. Folder names in the location are separated by '/'. Example location for a Datastore Cluster: **MyDatastoreClusterFolderOne/MyDatastoreClusterFolderTwo**. ||
+| **DatastoreNames**| Mandatory | string[] | The names of the Datastores that should be located in the specified Datastore Cluster. ||
+
+## Description
+
+The resource is used to add Datastores to the specified Datastore Cluster.
```
Comment (dmilov) [inline Documentation/DSCResources/DatastoreClusterDatastore/DatastoreClusterDatastore.md:17]: Why only add operation? It is not exactly a DSC resource convention

Comment (SimeonGerginov) [reply Documentation/DSCResources/DatastoreClusterDatastore/DatastoreClusterDatastore.md:17]: Renamed the DSC Resource to **DatastoreClusterAddDatastore** to better understand the covered use case.

File: Source/VMware.vSphereDSC/DSCResources/DatastoreClusterDatastore.ps1 (hunk 1 above)
Comment (SimeonGerginov) [reply Source/VMware.vSphereDSC/DSCResources/DatastoreClusterDatastore.ps1:101]: Renamed the DSC Resource to **DatastoreClusterAddDatastore** to better understand the covered use case.

**Pull request merged**